### DISCUSSION TITLE
Vlang code completion with v-analyzer not filtering on keystroke

### DISCUSSION THREAD

**Discussion by ghost** [helix-editor/helix#8067] [Troubleshooting] [answered]
### Summary

I'm using the `v-analyzer` language server for Vlang coding and when I type a `.` to get code completion options, the list does not get filtered on subsequent keystrokes. As the list is also not sorted, I have to either tab my way through the list looking for what I want, or just go look it up in the documentation. I believe this is a helix issue because the funcitonality works as expected in VSCode (so I am under the impression it's not a problem with `v-analyzer`).

### Reproduction Steps

```bash
$ cd /tmp
$ v new lstest
$ cd lstest
$ hx src/main.v
```
After `module` statement, type

`import os`

Then move to the `main` function and type `os.`. Get an unordered list of options. Press any key and the list goes away.

### Helix log

[helix-lstest.log](https://github.com/helix-editor/helix/files/12442996/helix-lstest.log)

### Platform

macOS Ventura 13.4.1

### Terminal Emulator

kitty 0.29.2

### Helix Version

helix 23.05 (7f5940be)

**Comment by the-mikedavis** [comment, marked as answer]
`v-analyzer` is sending the empty string as each completion item's `filterText`, so any keypress filters out every completion item. This is an upstream `v-analyzer` bug: it should omit the `filterText` field if it wants the client (editor) to use the `label` for filtering. See the `filterText` item under `CompletionItem`: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#completionItem

I suspect that this works in vscode because `""` is falsy in JavaScript, so it acts as though `completionItem.filterText` is undefined.

**Reply by ghost** [reply]
@the-mikedavis Thanks so much for the reply. I passed the information on to folks involved with `v-analyzer`.
